### PR TITLE
feat: update task

### LINKED ISSUE #206: Update task name
## Description

Implement a logic to update the task name on the edit page.

## Criteria

- [ ] Updating a record on `task` table

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Implement a logic to update the task name and description on the edit page.

## Changes

- Updating a record of the title on `task` table
- Updating a record of the description on `task` table
- Return an error if there is no data to update

## Review points

1. Set Content-Type: application/json on Headers tab in Postman
2. Send a **PATCH** method with the following data after selecting raw on Body tab
```
{
    "title": "update-title",
    "description": "update-desc"
}
```
3. Check whether the record is updated in `task` table

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/routes/task.route.ts [hunk 1]
```diff
@@ -9,10 +9,46 @@ const app = new Hono();
 
 export default app;
 
-app.patch('/:taskId', (c) => {
-  const taskId = c.req.param('taskId');
-  return c.json({ message: `Updating task id: ${taskId}` });
-});
+app.patch(
+  '/:taskId',
+  zValidator('json', taskCreationSchema.omit({ categoryId: true }).partial()),
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/task.route.ts]: I'd like you to define `taskCreationSchema.omit({ categoryId: true }).partial()` in the `src/constants/schema.ts` and use it by importing it from there. This makes it possible to reuse it in the frontend.

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you check my comments?

File: src/app/api/[[...route]]/routes/task.route.ts (hunk 1 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/task.route.ts]: I've created a new schema for task update

**Review by akiomatic — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**